torch/_decomp/__init__.py

```diff
@@ -250,13 +252,172 @@ def remove_decompositions(
+
+    cia_ops = set()
+    for op in torch.ops.aten:
+        op_packet = getattr(torch.ops.aten, op)
+        for overload in op_packet.overloads():
+            op_overload = getattr(op_packet, overload)
+            if _check_valid_to_preserve(op_overload) and _is_cia_op(op_overload):
+                cia_ops.add(op_overload)
+    return cia_ops
+
+
 # See NOTE [Core ATen Ops]
 #
 # list was copied from torch/_inductor/decomposition.py
 # excluding decompositions that results in prim ops
 # Resulting opset of decomposition is core aten ops
 def core_aten_decompositions() -> Dict[torch._ops.OperatorBase, Callable]:
+    decomp_table = _core_aten_decompositions_post_autograd()
+
+    # If it is fbcode change, we return the old decomposition list
+    from torch._inductor import config
+
+    if config.is_fbcode():
+        return decomp_table
+
+    aten = torch.ops.aten
+
+    # We are deleting custom decomp in core_aten_decomp
+    # for CIA ops but it should be fine technically
+    # because this table is only meant to be used in export context
+    # in which we really carefully control the decomp behaviour
+    # In any case, C++ decomps should be preferred
+    cia_ops_that_should_be_removed = [
+        aten.all.dimname,
+        aten.index_add.dimname,
+        aten.index_copy.dimname,
+        aten.index_fill.Dimname_Scalar,
+        aten.index_fill.Dimname_Tensor,
+        aten.norm.names_ScalarOpt_dim_dtype,
+        aten.norm.names_ScalarOpt_dim,
+        aten.silu_backward.default,
+        aten.std.default,
+        aten.std.dim,
+        aten.std.names_dim,
+        aten.std.correction_names,
+        aten.std_mean.default,
+        aten.std_mean.dim,
+        aten.std_mean.names_dim,
+        aten.std_mean.correction_names,
+        aten.upsample_bilinear2d.vec,
+        aten.upsample_trilinear3d.vec,
+    ]
+
+    for k in list(decomp_table.keys()):
+        if k in cia_ops_that_should_be_removed:
+            del decomp_table[k]
+
+    for op in _collect_all_valid_cia_ops():
+        # [NOTE] Seperating out func.decompose
+        # Ideally we should be able to just register func.decompose but
+        # we can't as this decomp is gonna be registered to the py_impl.
+        # As a result it will infinitely recurse. So we first check if the op
+        # has py_impl entry for CIA and if it is we use that first. If not,
+        # we register C++ query to py_impl.
+        if (
+            torch._C.DispatchKey.CompositeImplicitAutograd in op.py_kernels
+        ) and not isinstance(
+            op.py_kernels[torch._C.DispatchKey.CompositeImplicitAutograd],
+            torch._C.DispatchKey,
+        ):
+            decomp_table[op] = op.py_kernels[
+                torch._C.DispatchKey.CompositeImplicitAutograd
+            ]
+        else:
+            decomp_table[op] = partial(_is_special_op_to_decompose, kernel=op)
+    return decomp_table
+
+
+def decomp_table_to_post_autograd_aten():
```

Comment:
same with this - is the idea that we need this to stick around permanently? Or that we want to migrate everyone off of the old decomposition defaults, to either the new default or core_aten, and we want to kill this later

(if it's temporary, I would probably mark the API as private and add some comments explaining)

Reply:
Hmm i guess temporary, do  you see any reason it could be a permanent thing? I want to have it here so that for anyone who got annoyed at us for breaking their code, we just give them this table.

Reply:
nope, I don't think we want it to be permanent. Are you planning on going through internal and figuring out how to delete usages of it aftwards?